host/distro/devtree.go

```diff
@@ -35,11 +40,34 @@ func DTCompatible() []string {
 	return dtCompatible
 }
 
+// DTRevision returns the device revision (e.g. a02082 for the Raspberry Pi 3)
+// from the Linux device tree, or 0 if the file is missing or malformed.
+func DTRevision() uint32 {
+	mu.Lock()
+	defer mu.Unlock()
+
+	if dtRevisionRead {
+		return dtRevision
+	}
+	dtRevisionRead = true
+	b, err := ioutil.ReadFile("/proc/device-tree/system/linux,revision")
```

Comment:
Thinking about it more, this is simpler. ReadFile() won't return b if an error occurred, so there's no point in checking the error.

```
if b, _ := ioutil.ReadFile("/proc/device-tree/system/linux,revision"); len(b) >= 4 {
  dtRevision = binary.BigEndian.Uint32(b[:4])
}
return dtRevision
```

Reply:
Done.